.github/workflows/commitlint.yml

```diff
@@ -1,18 +0,0 @@
-name: "Commit Linter"
-on: pull_request
-jobs:
-  lint-commits:
-    runs-on: ubuntu-latest
-    env:
-      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-    steps:
-      - uses: actions/checkout@v2.3.1
-        with:
-          fetch-depth: 0
-      - run: npm install --save-dev @commitlint/cli
-      - run: |
-          if [[ "$(gh pr view ${{ github.event.pull_request.number }} --json commits --jq '.[][0].messageHeadline')" == vim-patch* ]];then
-            npx commitlint --from HEAD~1 --to HEAD --verbose --help-url https://github.com/neovim/neovim/blob/master/CONTRIBUTING.md#commit-messages --config .github/workflows/commitlint.config_patch.js
```

Comment:
wasn't expecting to remove the GHA node stuff, but can't say i don't like it... however we should keep the separate job (which can just call `make commitlint`); we don't want to fail other CI jobs

Reply:
Sounds good. I'll focus on getting it to work locally first.

Reply:
Haven't decided yet, but I may split up this PR into two parts if the PR becomes too big. In that case I'll unremove the current commitlinter workflow and only focus on making the manual linter work locally and then focus on the CI in part 2.

Reply:
> I'll unremove the current commitlinter workflow and only focus on making the manual linter work locally and then focus on the CI in part 2.

+1